index.js

```diff
@@ -27,100 +22,80 @@ function readFile(input) {
-      cenaStvarno =
-        cenaStvarno +
-        parseFloat(a[i].split(",")[4]) * parseFloat(a[i].split(",")[6]);
-    }
-    localStorage.setItem("baza", JSON.stringify(Baza));
-    localStorage.setItem("lager", JSON.stringify(cenaLager));
-    localStorage.setItem("lagerStvarno", JSON.stringify(cenaStvarno));
+    records.forEach(record => {
+      let obj = parseRecord(record);
+      storage[obj.id] = obj;      
+      actualPrice += obj.price * obj.realStock;
+      stockPrice += obj.price * obj.stock;
+    });
+    updateStorage();
   };
 
   reader.onerror = function () {
     console.log(reader.error);
   };
 }
 
+function updateStorage() {
+  localStorage.setItem("baza", JSON.stringify(storage));
+  localStorage.setItem("lager", stockPrice);
+  localStorage.setItem("lagerStvarno", actualPrice);
+}
+
+// parse csv string line to an actual object
+function parseRecord(csvRecord) {
+  let record = csvRecord.split(",");
+  return {
+    id: parseInt(record[0]),
+    name: record[1],
+    // not sure something: record[2],
+    // not sure something: record[3],
+    price: parseFloat(record[4]),
+    // not sure something: record[5]
+    realStock: parseFloat(record[6]),
+    stock: parseFloat(record[7]),
+    originalCSV: csvRecord,
+  }
+}
+
 window.addEventListener("DOMContentLoaded", () => {
   if (localStorage.getItem("baza")) {
-    Baza = JSON.parse(localStorage.getItem("baza"));
-    tekst.innerText = Baza.filter((x)=>x!==null).map(e=>(e.split(',').slice(0,2)).join('  ')).join("\r\n");
-    cenaLager = JSON.parse(localStorage.getItem("lager"));
-    cenaStvarno = JSON.parse(localStorage.getItem("lagerStvarno"));
+    storage = JSON.parse(localStorage.getItem("baza"));
+    text.innerText = storage.filter(x => x).map(e => `${e.id}  ${e.name}`).join("\r\n");
+    stockPrice = JSON.parse(localStorage.getItem("lager"));
+    actualPrice = JSON.parse(localStorage.getItem("lagerStvarno"));
   } else izborDiv.style.display = "block";
 });
 
 function kazi(e) {
-  const odgovor = Baza[e];
-  if (!Baza[e]) {
+  const odgovor = storage[e];
+  if (!storage[e]) {
     alert("Nemate takav proizvod u bazi");
   } else {
-    kol.focus();
-    document.getElementById("artikl").innerText = odgovor;
+    amount.focus();
+    document.getElementById("article").innerText = odgovor;
   }
 }
 
-prodajaBtn.addEventListener("click", () => {
+function styleMenuNav(clickedId) {
+  // no need to execute this each time in a loop
+  menuNav[clickedId].style.color = "white";
   for (let i = 0; i < 5; i++) {
-    meniNav[i].style.color = "silver";
-    meniNav[0].style.color = "white";
-  }
-  for (let j = 0; j < forma.length; j++) {
-    forma[j].style.display = "none";
-  }
-
-  prodaja();
-});
-
-nabavkaBtn.addEventListener("click", () => {
-  for (let i = 0; i < 5; i++) {
-    meniNav[i].style.color = "silver";
-    meniNav[1].style.color = "white";
+    if (clickedId != i) {
+      menuNav[i].style.color = "silver";
+    }
```

Comment:
Primetio sam da ima 4 iste funkcije, koje su previse slicne, pa sam to izvuko u jednu funkciju, dosta skracuje kod.

Reply:
tacno tako. zurio sam i nisam stigao da skontam kako da ostavim samo clicked da ostane beo,, tacnije nisam uopste razmisljao da uhvatim id iz niza button-a